RIOT-OS/RIOT
# pkg/tinyusb: add usb_board_reset support

### Contribution description

This PR provides tinyUSB support for module `usb_board_reset` when `stdio_tinyusb_cdc_acm` is used as stdio. The PR also provides some compilation fixes that occurred when adding `usb_board_reset` support and that are needed for successful compilation. In detail:
- 6aa267c4b57a10ed59384d77e913f3757bca71f5 and d390b6c7f6d79613f7967e3434585a310c0f9119 add the tinyUSB support of `usb_board_rest`.
- ed585cfc2225b6526ef993eebc8b1fff16276419 fixes a compilation problem when `NDEBUG` is defined.
- b97da1bfdee551e728540c48cdc7123865125a26 fixes a compilation problem when compiling tinyUSB for an application where `USB_VID`/`USB_PID` are not defined.
- 4b328203d25f3abac9eb0f61d7ff3e837f6d5fed is a very small cleanup of `tests/pkg_tinyusb_cdc_msc/app.config.test` (out of scope of this PR but too small to fix as separate PR)

### Testing procedure

Use any board that provides the `tinyusb_device` feature, e.g. the `stm32f429i_disc1` and flash:
```
USEMODULE=' stdio_tinyusb_cdc_acm riotboot_reset usb_board_reset' \
BOARD=stm32f429i-disc1 make -C tests/shell flash
```
Once the USB port is recognized, execute
```
USEMODULE='riotboot_reset usb_board_reset stdio_tinyusb_cdc_acm' \
BOARD=stm32f429i-disc1 make -C tests/shell term PORT=/dev/ttyACM1
```
When using command
```
stty -F /dev/ttyACM1 raw ispeed 600 ospeed 600 cs8 -cstopb ignpar eol 255 eof 255
```
nothing should happen without commit 59e8ae837a00169a9f140e7a946823cecaa2624e, but the board should restart with commit 59e8ae837a00169a9f140e7a946823cecaa2624e. Without the other commits, the compilation fails completely.

Another example for testing would be:
```
FEATURES_PROVIDED=tinyusb_device USEMODULE='usb_board_reset stdio_tinyusb_cdc_acm' \
BOARD=nrf52840-mdk-dongle make -C tests/shell
```
When using command 
```
stty -F /dev/ttyACM1 raw ispeed 600 ospeed 600 cs8 -cstopb ignpar eol 255 eof 255
```
the node restarts the application. With command
```
stty -F /dev/ttyACM0 raw ispeed 1200 ospeed 1200 cs8 -cstopb ignpar eol 255 eof 255
```
the node restarts in bootloader mode.

### Issues/PRs references

Prerequisite for PR #18998
